src/tree/contexts/ContextTreeItem.ts

```diff
@@ -60,6 +61,18 @@ export class ContextTreeItem extends AzExtTreeItem {
     }
 
     public async use(context: IActionContext): Promise<void> {
-        return dockerContextManager.use(this.name);
+        try {
+            ext.isContextSwitchInProgress = true;
+            return this.runWithTemporaryDescription(
+                localize('vscode-docker.tree.context.using', 'Using...'),
+                async () => {
+                    await dockerContextManager.use(this.name);
+                    // The refresh logic will use the cached contexts retrieved by polling (auto refresh logic)
+                    // Since the context is changed, clear the cached contexts, otherwise the old context will be shown as active
+                    ext.contextsRoot.clearPollingCache();
```

Comment:
Agreed, less computation cost.

Reply:
Without clearing the cache, the auto refresh will use the old cached value, so we must clear the cache. I am reluctant to modifying the view model which require access to the old current and it is better to always reflect the actual data coming from backend.

Reply:
It is fine to clear the cache, although `hasChanged()` will do so automatically anyway next time it is called. My point is that after `dockerContextManager.use()` we **know** that the view model is invalid and what the right view model should be, so no need to wait for the refresh process clean things up, we can make the change immediately. This results in better UI responsiveness.

Also, with this change the `ContextTreeItem` now "knows" about the refresh & data caching logic in the root tree item base class. This increases coupling and makes things more error prone.

Reply:
The cache must be cleared because of the way how the refresh logic works in `LocalRootTreeItemBase`. The specific scenario is
1. Auto refresh caches the context list as part of auto refresh. The next call to `hasChanged()` will happen after 20 seconds.
2. Now user switches context and we invoke refresh.
3. Without clearing the cache, the refresh will use the cached context and will not show the latest current context.

Manipulating the view model require manipulating both current and previous current node and also require access to `RefreshUI` which is not exposed now. Also require copying the loading icon to this extension. In short, we need to duplicate what runWithTemporaryDescription does in the base class and it gets complex and it is not worth.

To make the `ContextTreeItem` not know about the refresh logic, I moved the code to `userDockerContext`